packages/retail-ui-extensions/src/extension-points/extension-points.ts

```diff
@@ -0,0 +1,21 @@
+import {StandardApi} from '../extension-api';
+import {RenderExtension} from './render-extension';
+
+type Components = typeof import('../components');
+type AllComponents = Components[keyof Components];
+
+type Tile = typeof import('../components/Tile');
+type TileComponent = Tile[keyof Tile];
+
+export interface ExtensionPoints {
+  'Retail::SmartGrid::Tile': RenderExtension<
+    StandardApi<'Retail::SmartGrid::Tile'>,
+    TileComponent
+  >;
+  'Retail::SmartGrid::Modal': RenderExtension<
```

Comment:
Out of curiosity, is the intention to have extensions that target one of these two extension points, or do you expect most/ all extensions will target both? If this is to create a "click on tile opens up modal"-type of workflow, I'd probably recommend making the modal be a component you render from the tile extension point, instead of a dedicated extension point. But I'm just guessing at what you're trying to achieve with this (a brief comment in code about the use cases for these extension points is probably a good idea).

Reply:
Great question! The tile could open the full screen modal, or it could execute another action. I think the modal does need to be its own extension point as there is no guarantee it will only be presented modally (it could be pushed onto the stack in cart actions as well). In all cases, on the client side the "Modal" is a full on react native screen that needs to be navigated to, so the best way to handle this is by having it be its own extension point.

Reply:
There doesn't need to be any direct connection between extension points and views in your application. The extension can live entirely outside the routing hierarchy, and you should be able to interpret the components rendered by the extension into the appropriate React component/ routing structure in the host application. Separating extensions across multiple extension points introduces complexity for the developer because there is no natural mechanism for sharing state between them (other than global state, which is generally smelly in JavaScript).

A single extension point lets extensions do a lot of powerful things, like share state, share React context, preload data for views that aren't needed just yet, etc. If you need extensions to be able to target each extension point individually, it makes sense to keep them separate, but if they are part of a single UI "flow", I strongly recommend keeping them as one extension point and using components to express the views getting pushed on the stack.